src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/CountFalsePositives.java

```diff
@@ -32,62 +30,101 @@
 /**
  * Created by tsato on 12/28/16.
  */
-public class CountFalsePositives extends CommandLineProgram {
-    @Argument(doc = "a line-delimited list of mutect2 vcf files", fullName= "", shortName = "V", optional = false)
-    protected File vcfList;
 
-    @Argument(doc = "output file", fullName= "output", shortName = "O", optional = false)
-    protected File output;
+/** TODO: this class should be a variant walker. That way we can ask questions like how are the false positive rates
+ *  over these intervals and so on. For now, we look at the entire vcf and assume ICE intervals
+ */
+
+public class CountFalsePositives extends VariantWalker {
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "Output file (if not provided, defaults to STDOUT)", common = false, optional = true)
+    private File outputFile = null;
+
+    private PrintStream outputStream = null;
+
+    long targetTerriroty = 0;
+    long indelFalsePositiveCount = 0;
+    long snpFalsePositiveCount = 0;
+    String id;
+
+
 
     // TODO: eventually use tumor and normal sample names instead of the file name. To do so we must extract them from the vcf, which I don't know how.
     @Override
-    public Object doWork() {
-        Utils.regularReadableUserFile(vcfList);
-        try ( FalsePositiveTableWriter writer = new FalsePositiveTableWriter(output) ) {
-            List<String> vcfFilePaths = Files.readAllLines(Paths.get(vcfList.toString()));
-            List<FalsePositiveRecord> falsePositiveRecords = vcfFilePaths.stream().map(str -> new FalsePositiveRecord(new File(str))).collect(Collectors.toList());
-            writer.writeAllRecords(falsePositiveRecords);
+    public void onTraversalStart() {
+        try {
+            outputStream = outputFile != null ? new PrintStream(outputFile) : System.out;
+        }
+        catch ( final FileNotFoundException e ) {
+            throw new UserException.CouldNotReadInputFile(outputFile, e);
+        }
+
+        List<SimpleInterval> intervals =  intervalArgumentCollection.getIntervals(getReferenceDictionary());
+        targetTerriroty = intervals.stream().mapToInt(i -> i.size()).sum();
+
+        // TODO: ideally we would identify the tumor sample name and normal sample name from the vcf header but
+        // there doesn't seem to be a function that let us query a sample name by its ID (e.g. Tumor/Normal) in VCFHeader class.
+        id = FilenameUtils.getBaseName(drivingVariantFile);
+    }
+
+    @Override
+    public void apply(final VariantContext variant, final ReadsContext readsContext, final ReferenceContext referenceContext, final FeatureContext featureContext) {
+        if (variant.isFiltered()) return;
+
+        if (variant.isIndel()) {
+            indelFalsePositiveCount++;
+        } else {
+            snpFalsePositiveCount++;
+        }
```

Comment:
Hmm didn't know about `Mixed.` I looked into it a bit and it looks like a mixed variatn doesn't necessarily need to be a SNP and INDEL. Could be two INDELs, even two SNPs? More than two alleles? @vdauwera do you know?